lib/utils.js

```diff
@@ -115,7 +115,7 @@ utils.getNamesFromReq = function getNamesFromReq(request, bucketNameFromHost) {
             resources.bucket = urlArr[1];
             resources.object = urlArr.slice(2).join('/');
         } else if (urlArr.length === 1) {
-            resources.bucket = urlArr[1];
+            resources.bucket = urlArr[0];
```

Comment:
Why have you changed that? 
For exemple if path == '/mybucket'  so, resources.bucket == ''
No?